api/v1alpha1/contentlibraryitemimportrequest_types.go

```diff
@@ -30,20 +30,46 @@ const (
 
 // ContentLibraryItemImportRequestSource contains the specification of the source for the import request.
 type ContentLibraryItemImportRequestSource struct {
+	// +required
+
 	// URL is the endpoint that points to a file that is to be imported as a new Content Library Item in
 	// the target vSphere Content Library. If the target item type is ContentLibraryItemTypeOvf, the URL
-	// should point to an OVF descriptor file (.ovf) or an OVA file (.ova), otherwise the SourceValid
-	// condition will become false in the status.
-	// +required
+	// should point to an OVF descriptor file (.ovf), an OVA file (.ova), or an ISO file (.iso). Otherwise,
+	// the SourceValid condition will become false in the status.
 	URL string `json:"url"`
 
+	// +optional
+
 	// PEM encoded SSL Certificate for this endpoint specified by the URL. It is only used for HTTPS connections.
 	// If set, the remote endpoint's SSL certificate is only accepted if it matches this certificate, and no other
 	// certificate validation is performed.
 	// If unset, the remote endpoint's SSL certificate must be trusted by vSphere trusted root CA certificates,
 	// otherwise the SSL certification verification may fail and thus fail the import request.
-	// +optional
 	SSLCertificate string `json:"sslCertificate,omitempty"`
+
+	// +optional
+
+	// Checksum contains the checksum algorithm and value calculated for the
+	// file specified in the URL. If omitted, the checksum will not be verified.
+	Checksum *Checksum `json:"checksum,omitempty"`
+}
+
+// Checksum contains the checksum value and algorithm used to calculate that
+// value.
+type Checksum struct {
+	// +optional
+	// +kubebuilder:validation:Enum=SHA1;SHA256;SHA512;MD5
+	// +kubebuilder:default=SHA256
+
+	// Algorithm is the algorithm used to calculate the checksum. Supported
+	// algorithms are "SHA1", "SHA256", "SHA512", and "MD5".
```

Comment:
Though CLS still supports MD5 / SHA1, they are considered weak / vulnerable, consider only supporting the stronger SHA256, SHA512 algorithms, do we have specific use cases to support MD5/SHA1?

Reply:
No specific use case but I wanted to keep them as is to support all algorithms that CLS provides (in case users want to specify MD5/SHA1 for speed validation). Also, govc supports all these algorithms so it's nice to keep it consistent here?

Reply:
Yeah, I feel this is a new API, though it's backed by the CL API, it doesn't have to support all the options that CL API provides, especially for the options that may eventually be removed. If there are valid customer use cases that require those old algorithms, we can add them later.

Reply:
Sounds fair, removed SHA1 and MD5 as supported values.